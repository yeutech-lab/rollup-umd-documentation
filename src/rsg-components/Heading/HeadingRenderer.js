import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Styled from 'rsg-components/Styled';

const styles = ({ color, fontFamily, fontSize }) => ({
	heading: {
		margin: 0,
		color: color.base,
		fontFamily: fontFamily.base,
		fontWeight: 'normal',
	},
	heading1: {
		fontSize: fontSize.h1,
	},
	heading2: {
		fontSize: fontSize.h2,
	},
	heading3: {
		fontSize: fontSize.h3,
	},
	heading4: {
		fontSize: fontSize.h4,
	},
	heading5: {
		fontSize: fontSize.h5,
	},
	heading6: {
		fontSize: fontSize.h6,
	},
});

function HeadingRenderer({ classes, level, children, ...props }) {
	const Tag = `h${level}`;
	const headingClasses = cx(classes.heading, classes[`heading${level}`]);

	return (
		<Tag {...props} className={headingClasses}>
			{children}
		</Tag>
	);
}

HeadingRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
	children: PropTypes.node,
};

export default Styled(styles)(HeadingRenderer);
