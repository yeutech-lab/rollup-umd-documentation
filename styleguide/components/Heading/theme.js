/**
 * BOOTSTRAP THEME STYLEGUIDE - HEADING TAG
 *
 * this file is not meant to be used and must be kept as original
 * @param userTheme
 */
export const makeTheme = (userTheme = {}) => {
  /* eslint dot-notation: 'off', new-cap: 'off' */
  // Variables
  //
  // Copy settings from this file into the provided `_custom.scss` to override
  // the Bootstrap defaults without modifying key, versioned files.
  const v = {};
  const u = userTheme;

  v['$rsg-heading-margin'] = u['$rsg-heading-margin'] || 0;
  v['$rsg-heading-color'] = u['$rsg-heading-color'] || '#f44842';
  v['$rsg-heading-fontFamily'] = u['$rsg-heading-fontFamily'] || [
    '-apple-system',
    'BlinkMacSystemFont',
    '"Segoe UI"',
    '"Roboto"',
    '"Oxygen"',
    '"Ubuntu"',
    '"Cantarell"',
    '"Fira Sans"',
    '"Droid Sans"',
    '"Helvetica Neue"',
    'sans-serif',
  ];
  v['$rsg-heading-fontWeight'] = u['$rsg-heading-fontWeight'] || 'normal';
  v['$rsg-heading-h1-font-size'] = u['$rsg-heading-h1-font-size'] || 55;
  v['$rsg-heading-h2-font-size'] = u['$rsg-heading-h2-font-size'] || 45;
  v['$rsg-heading-h3-font-size'] = u['$rsg-heading-h3-font-size'] || 30;
  v['$rsg-heading-h4-font-size'] = u['$rsg-heading-h4-font-size'] || 20;
  v['$rsg-heading-h5-font-size'] = u['$rsg-heading-h5-font-size'] || 18;
  v['$rsg-heading-h6-font-size'] = u['$rsg-heading-h6-font-size'] || 16;

  return Object.assign({}, u, v);
};

export default makeTheme();
