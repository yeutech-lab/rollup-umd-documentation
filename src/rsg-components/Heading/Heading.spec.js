import React from 'react';
import { html } from 'cheerio';
import Heading from './index';

describe('Heading', () => {
	it('should render a heading according to the level', () => {
		let actual = shallow(<Heading level={3}>The heading</Heading>);
		expect(actual.dive().name()).toBe('h3');

		actual = shallow(<Heading level={5}>The heading</Heading>);
		expect(actual.dive().name()).toBe('h5');
	});

	it('should render a heading', () => {
		const actual = render(<Heading level={2}>The heading</Heading>);

		expect(html(actual)).toMatchSnapshot();
	});
});
