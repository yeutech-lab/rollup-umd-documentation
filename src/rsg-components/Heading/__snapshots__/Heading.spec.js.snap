// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render a heading 1`] = `

<h2 class="rsg--heading-0 rsg--heading2-2">
  The heading
</h2>

`;
